src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -1105,10 +1107,12 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                         $message = ''
                                         $prefix = ''
 
-                                        # Don't show line information if script module
-                                        if (($myinv -and $myinv.ScriptName -or $myinv.ScriptLineNumber -gt 1 -or $err.CategoryInfo.Category -eq 'ParserError') -and !($myinv.ScriptName.EndsWith('.psm1', [System.StringComparison]::OrdinalIgnoreCase))) {
+                                        # The checks here determine if we show line detailed error information:
+                                        # - check if `ParserError` and comes from PowerShell which eventually results in a ParseException, but during this execution it's an ErrorRecord
+                                        # - check if invocation is a script or multiple lines in the console
+                                        # - check that it's not a script module as expectation is that users don't want to see the line of error within a module
+                                        if (((($err.CategoryInfo.Category -eq 'ParserError' -and $err.Exception -is 'System.Management.Automation.ParentContainsErrorRecordException') -or $myinv.ScriptName -or $myinv.ScriptLineNumber -gt 1)) -and !($myinv.ScriptName -match '\.psm1$')) {
```

Comment:
It looks like you add one extra pair of parentheses.
```suggestion
                                        if ((($err.CategoryInfo.Category -eq 'ParserError' -and $err.Exception -is 'System.Management.Automation.ParentContainsErrorRecordException') -or $myinv.ScriptName -or $myinv.ScriptLineNumber -gt 1) -and !($myinv.ScriptName -match '\.psm1$')) {
```